### ISSUE TITLE
Race condition in fenix/update-android-components

### ISSUE THREAD

**Issue by st3fan** [mozilla-mobile/relbot#20]
This is an odd one .. we check and find out that the current A-C version in Fenix is 67.0.9 (from `AndroidComponents.kt` but then when we update `AndroidComponents.kt` to 67.0.10 we fail because it already was 67.0.10.

Race condition? I checked the PR before this happened and it was definitely landed/merged.

```
2020-12-08 14:14:13.872002 Looking at Fenix Beta on releases/v84.0.0
2020-12-08 14:14:13.999420 Current A-C version in Fenix is 67.0.9
2020-12-08 14:14:14.139332 Latest A-C version available is 67.0.10
2020-12-08 14:46:57.505593 We should upgrade Fenix 84 Beta to Android-Components 67.0.10
2020-12-08 14:46:57.819619 Last commit on releases/v84.0.0 is ee505c9f03d15dd914f979ca2dbc9c6676713238
2020-12-08 14:46:57.819657 Creating branch relbot/AC-67.0.10 on ee505c9f03d15dd914f979ca2dbc9c6676713238
2020-12-08 14:46:58.456664 Updating AndroidComponents.kt from 67.0.9 to 67.0.10 on relbot/AC-67.0.10
Traceback (most recent call last):
  File "relbot.py", line 97, in <module>
    main(sys.argv, ac_repo, fenix_repo, author, debug)
  File "relbot.py", line 58, in main
    fenix.update_android_components(ac_repo, fenix_repo, author, debug)
  File "/Users/stefan/Mozilla/relbot/src/fenix.py", line 61, in update_android_components
    _update_ac_version(fenix_repo, pr_branch_name, current_ac_version, latest_ac_version, author)
  File "/Users/stefan/Mozilla/relbot/src/fenix.py", line 14, in _update_ac_version
    raise Exception("Update to AndroidComponents.kt resulted in no changes: maybe the file was already up to date?")
Exception: Update to AndroidComponents.kt resulted in no changes: maybe the file was already up to date?
```

**Comment by st3fan** [comment]
Oh! Why is there half an hour between these two lines!?

```
2020-12-08 14:14:14.139332 Latest A-C version available is 67.0.10
2020-12-08 14:46:57.505593 We should upgrade Fenix 84 Beta to Android-Components 67.0.10
```

The PR landed at 14:39 so that definitely explains things.

**Comment by st3fan** [comment]
Tempted to close this because things worked as expected: we raised an error when things were not what we were expecting.

We do need to clean up better because the PR branch was created. But there is #4 for that.
